### DISCUSSION TITLE
Java Projects no more show Src folder and instead shows "Source PAckages is owned by project java, cannot be used here, see issue #138310 for details"

### DISCUSSION THREAD

**Discussion by thevpc** [apache/netbeans#8074] [General]
### Apache NetBeans version

Apache NetBeans 24

### What happened

After opening a java maven project , netbeans shows no more source folder and behaves oddly
Here is the message shown

![image](https://github.com/user-attachments/assets/d430a77b-d371-4905-8ef5-0b86a74c2b32)

Log file does not provide any clue.
The only thing i see is 
WARNING [org.netbeans.modules.java.source.indexing.JavaIndex]: Ignoring root with no ClassPath:  ...
whereas the folders belong to projects with existing dependencies
Besides when I creat e a new java project, here is what it looks
![image](https://github.com/user-attachments/assets/2021caa2-e61f-42c5-b266-a25f60d39572)

src folder is shown but twice!!

This happens randomly, and not always. But when it happens, i find no way to get it working properly but to delete completly netbeans cache and config folders.
Deleting only cache does not change anything. But i noticed that deleting Modules folder under config does!
this is the best workaround i can find for now.

### Language / Project Type / NetBeans Component

Java Maven Project with modules

### How to reproduce

I cannot tell exactly how it reproduces as it is random and
Randomly it shows the following...
![image](https://github.com/user-attachments/assets/d430a77b-d371-4905-8ef5-0b86a74c2b32)
This worked properly before nb22, on my machine, I was hoping it to be fixed but it didnt in 23 neither 24.

### Did this work correctly in an earlier version?

Apache NetBeans 21

### Operating System

Linux version 6.11.8-1-default running on amd64; UTF-8; en_US (nb)

### JDK

21.0.5; OpenJDK 64-Bit Server VM 21.0.5+11-suse-1.1-x8664

### Apache NetBeans packaging

Apache NetBeans binary zip

### Anything else

It is not systematic, but once it happens, there is no way to fix it, i always try to delete config and cache folder.

### Are you willing to submit a pull request?

No

**Comment by matthiasblaesing** [comment]
Test with a clean userdir (`netbeans --userdir <path_to_an_not_yet_existing_directory>`) I bet your configuration became corrupted and that is the reason you are seeing this. You can start also with a complete clean user dir, if you delete the folders `$HOME/.netbeans/VERSION` and `$HOME/.cache/netbeans/VERSION.`

As this is not a general problem, I'll turn this into a discussion.

**Comment by thevpc** [comment]
Thank you for your response, but i am still encountering this issue few "clicks" after starting a new userdir. I have no plugins installed. I cannot tell exactly how it reproduces as it is random.

**Comment by IvoLimmen** [comment]
I am currently stuck with the same issue. Found this issue. I can't open 1 module. Tried to clean it but I used the snap version of Netbeans 28. I uninstalled the snap version and installed the DEB version. Nothing was in my `.netbeans` directory and/or in my `.cache/netbeans`. When I opened the project it still came back with the same issue. Plus: maven is totally fine with all of this.

**Comment by matthiasblaesing** [comment]
This needs a reproducer as in steps someone else has to take to reach the problematic state.

**Comment by IvoLimmen** [comment]
I understand but I can't share this project and building a demo project of 40 modules that might reach the same state, I don't have that kind of time.
But I don't really get why Netbeans already shows this issue; someone reached this point already...
